### PR TITLE
Docs: Add style guide for commit messages

### PR DESCRIPTION
**Pull request by raphaelm**

### CODE REVIEW

**Review by pc-coholic — APPROVED** [review]

File: doc/development/contribution/style.rst [hunk 1]
```diff
@@ -20,8 +25,62 @@ Coding style and quality
...
+-------------------------
+
+
+
+Most commits should start as pull requests, therefore this applies to the titles of pull requests as well since
+the pull request title will become the commit message on merge. We prefer merging with GitHub's "Squash and merge"
+feature if the PR contains multiple commits that do not carry value to keep. If there is value in keeping the
+individual commits, we use "Rebase and merge" instead. Merge commits should be avoided.
+
+* The commit message should start with a single subject line and can optionally followed by a commit message body.
```
Comment (wiffbi) [inline doc/development/contribution/style.rst:38]:
```suggestion
* The commit message should start with a single subject line and can optionally be followed by a commit message body.
```

File: doc/development/contribution/style.rst [hunk 2]
```diff
@@ -20,8 +25,62 @@ Coding style and quality
...
+
+  * The subject line should be the shortest possible representation of what the commit changes. Someone who reviewed
+    the commit a couple of weeks ago should able to immediately remember the commit based on the subject line and tell
```
Comment (wiffbi) [inline doc/development/contribution/style.rst:41]:
```suggestion
  * The subject line should be the shortest possible representation of what the commit changes. Someone who reviews
    the commit in a couple of weeks should able to immediately remember the commit based on the subject line and tell
```

**Review by wiffbi — COMMENTED** [review]
Thank you, lots of valuable info for me too!

File: doc/development/contribution/style.rst (hunk 2 above)
Comment (raphaelm) [reply doc/development/contribution/style.rst:41]: Hmm, that's not precisely what I meant. My sentiment was: If I'm the one reviewing the PR, and I'm the one writing the release notes four weeks later, I want the subject line to be all I need to recognize and remember the change.

Comment (wiffbi) [reply doc/development/contribution/style.rst:41]: Ah, now I get it! Maybe this way around makes it clearer?

```suggestion
  * The subject line should be the shortest possible representation of what the commit changes. Someone who reviewed
    the commit should able to immediately remember the commit in a couple of weeks based on the subject line and tell
```

**Review by wiffbi — APPROVED** [review]

**Pull request merged**